### PR TITLE
refactor fcntl29 with new API

### PR DESCRIPTION
**Pull request by ilmanzo**
Keeps same test logic, only use newer API with the goal of more clean and concise code.

### CODE REVIEW

File: testcases/kernel/syscalls/fcntl/fcntl29.c [hunk 1]
```diff
@@ -1,102 +1,49 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (Avinesh) [inline testcases/kernel/syscalls/fcntl/fcntl29.c:1]: As per original test this should be- `GPL-2.0`

File: testcases/kernel/syscalls/fcntl/fcntl29.c [hunk 2]
```diff
@@ -1,102 +1,49 @@
...
-#include <unistd.h>
-#include <string.h>
-#include <signal.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <pwd.h>
-
-#include "test.h"
-#include "safe_macros.h"
 #include "lapi/fcntl.h"
```
Comment (Avinesh) [inline testcases/kernel/syscalls/fcntl/fcntl29.c:15]: nit: we can remove this empty line.

**Review by Avinesh — COMMENTED** [review]
Rest looks good to me.

File: testcases/kernel/syscalls/fcntl/fcntl29.c [hunk 3]
```diff
@@ -1,102 +1,49 @@
...
 static void setup(void)
 {
-	tst_sig(NOFORK, DEF_HANDLER, cleanup);
-
-	tst_tmpdir();
-
-	TEST_PAUSE;
-
-	test_fd = SAFE_CREAT(cleanup, "testfile", 0644);
+	test_fd = SAFE_CREAT("testfile", 0644);
```
Comment (pevik) [inline testcases/kernel/syscalls/fcntl/fcntl29.c:22]: nit: `fd` would be enough (`test_fd` is too long).

File: testcases/kernel/syscalls/fcntl/fcntl29.c [hunk 4]
```diff
@@ -1,102 +1,49 @@
...
 {
 	if (test_fd > 0)
-		SAFE_CLOSE(NULL, test_fd);
+		SAFE_CLOSE(test_fd);
+}
 
-	tst_rmdir();
+static void run(void)
+{
+	TST_EXP_FD(fcntl(test_fd, F_DUPFD_CLOEXEC, 0), "fcntl test F_DUPFD_CLOEXEC");
```
Comment (pevik) [inline testcases/kernel/syscalls/fcntl/fcntl29.c:33]: Actually keeping the default output is more informative and shorter:
```
TST_EXP_FD(fcntl(test_fd, F_DUPFD_CLOEXEC, 0));
```

File: testcases/kernel/syscalls/fcntl/fcntl29.c [hunk 5]
```diff
@@ -1,102 +1,49 @@
...
+	int dup_fd = TST_RET;
+
+	TST_EXP_POSITIVE(fcntl(dup_fd, F_GETFD), "fcntl test F_GETFD");
+	if (TST_RET & FD_CLOEXEC)
+		tst_res(TPASS, "fcntl test F_DUPFD_CLOEXEC success");
```
Comment (pevik) [inline testcases/kernel/syscalls/fcntl/fcntl29.c:38]: nit: maybe slightly different wording + space:

````C
if (TST_RET & FD_CLOEXEC)
	tst_res(TPASS, "fcntl() set F_DUPFD_CLOEXEC");
else
	tst_res(TFAIL, "fcntl() did not set F_DUPFD_CLOEXEC");

SAFE_CLOSE(dup_fd);

File: testcases/kernel/syscalls/fcntl/fcntl29.c (hunk 2 above)
Comment (ilmanzo) [reply testcases/kernel/syscalls/fcntl/fcntl29.c:15]: well.. AFAIK It's removed on the commit; you are commenting the original `fcntl29.c` code

**Comment by pevik** [comment]
@Avinesh @ilmanzo FYI final changes before merge (use valid SPDX, also TPASS/TFAIL should mention `FD_CLOEXEC` therefore add it to docs).
@Avinesh Feel free to further correct my english.

```diff
diff --git testcases/kernel/syscalls/fcntl/fcntl29.c testcases/kernel/syscalls/fcntl/fcntl29.c
index 3491df062..24d3983ed 100644
--- testcases/kernel/syscalls/fcntl/fcntl29.c
+++ testcases/kernel/syscalls/fcntl/fcntl29.c
@@ -1,5 +1,4 @@
-// SPDX-License-Identifier: GPL-2.0
-
+// SPDX-License-Identifier: GPL-2.0-only
 /*
  * Copyright (c) 2014 Fujitsu Ltd.
  * Author: Xiaoguang Wang <wangxg.fnst@cn.fujitsu.com>
@@ -8,11 +7,11 @@
 
 /*\
  * [Description]
- * Basic test for fcntl(2) using F_DUPFD_CLOEXEC argument.
+ *
+ * Basic test for fcntl(2) using F_DUPFD_CLOEXEC and getting FD_CLOEXEC.
  */
 
 #include "lapi/fcntl.h"
-
 #include <tst_test.h>
 
 static int fd;
@@ -35,9 +34,9 @@ static void run(void)
 
        TST_EXP_POSITIVE(fcntl(dup_fd, F_GETFD), "fcntl test F_GETFD");
        if (TST_RET & FD_CLOEXEC)
-               tst_res(TPASS, "fcntl() set F_DUPFD_CLOEXEC");
+               tst_res(TPASS, "fcntl() set FD_CLOEXEC");
        else
-               tst_res(TFAIL, "fcntl() did not set F_DUPFD_CLOEXEC");
+               tst_res(TFAIL, "fcntl() did not set FD_CLOEXEC");
 
        SAFE_CLOSE(dup_fd);
 }

**Comment by Avinesh** [comment]
@pevik, @ilmanzo , looks good.
Feel free to add `Reviewed-by: Avinesh Kumar <akumar@suse.de>` while merging.

**Review by Avinesh — APPROVED** [review]

**Comment by acerv** [comment]
@pevik is there a reason why this ticket has been closed without merge?

**Pull request closed without merging**